File: level-minus1/m2-1-stopwatch/src/c/stopwatch.c
/**
 * @file     stopwatch.c
 * @brief    output elapsed 10s
 * @version  1.0
 * @author   Wayson
 * @date     2025.10.5
 **/

#include <stdio.h>
#include <time.h>

int main(int argc, char* argv[])
{
    (void)argc;
    (void)argv;

    struct timespec req = {1, 0};

    for (int i = 0; i < 10; i++)
    {
        printf("elapsed : %d s\n", i + 1);
        fflush(stdout);
        nanosleep(&req, NULL);
    }    

    return 0;
}

File: level-minus1/m2-1-stopwatch/src/py/stopwatch.py
#!usr/bin/python3
'''
@file     stopwatch.py
@brief    output elapsed 10s
@version  1.0
@author   Wayson
@date     2025.10.5
'''

import time

for i in range(0, 10):
    print(f'elapsed : {i + 1} s', end='', flush=True)
    print()
    time.sleep(1)